### DISCUSSION TITLE
Next middleware throwing strange error

### DISCUSSION THREAD

**Discussion by FlinnBurgess** [vitest-dev/vitest#5233] [Q&A] [answered]
I'm using NextJS, and am trying to switch from Jest to Vitest. I have all but one of my test files passing, but the final failure doesn't make much sense to me.

We use `next-compose-middleware` and have a test file specifically for testing our middleware. However when I try running the test file I get the following error:

```
Cannot find module '.../node_modules/next-compose-middleware/dist/compose' imported from .../node_modules/next-compose-middleware/dist/index.js
```

This error doesn't occur in Jest, and I'm not sure what is causing it. The files mentioned definitely exist. Does anyone have any suggestions?

**Comment by hi-ogawa** [comment, marked as answer]
I think this is `next-compose-middleware`'s ESM packaging issue. NodeJS ESM resolution requires exact extension to be rewritten out, so `./compose` needs to be `./compose.js` here https://unpkg.com/browse/next-compose-middleware@1.0.0/dist/index.js

I can reproduce the same error simply by:
https://stackblitz.com/edit/vitest-dev-vitest-pfqxst?file=repro.mjs

```js
// node repro.mjs
import * as lib from 'next-compose-middleware';
console.log(lib);
```

Probably It would be better to report the issue to https://github.com/kj455/next-compose-middleware, but workaround provided by Vite/Vitest is to put such non-esm-compatible package into `server.deps.inline` like following. Would this work for you?
https://stackblitz.com/edit/vitest-dev-vitest-pfqxst?file=vite.config.ts

```ts
export default defineConfig({
  test: {
    server: {
      deps: {
        inline: ['next-compose-middleware'],
      },
    },
  },
});
```

**Reply by FlinnBurgess** [reply]
Thanks for the reply, that did the job 😄 I'll let them know about the issue
